src/OsuSharp/Mapper/OsuSharpMapper.cs

```diff
@@ -52,7 +53,16 @@ internal static class OsuSharpMapper
             {
                 if (implementingProperty.PropertyType.IsEnum)
                 {
-                    modelValue = Enum.Parse(implementingProperty.PropertyType, modelValue.ToString()!, true);
+                    if (implementingProperty.PropertyType == typeof(TeamType))
+                    {
+                        modelValue = modelValue.ToString() == "head-to-head"
```

Comment:
I think we should add `Description` attribute to the enum fields to parse them more easily.

Reply:
In this case I like to use more appropriate `EnumMember`, but because `TeamType` is located in Domain we are coupling API contract to non-API part of the project

Reply:
So we might resolve it more fluently by registering related string->Enum value mappings, but it's just straight up AutoMapper